zerver/views/integrations.py

```diff
@@ -144,4 +144,9 @@ def integration_doc(request: HttpRequest, integration_name: str=REQ(default=None
 
     doc_html_str = render_markdown_path(integration.doc, context)
 
+    if not (settings.ROOT_DOMAIN_LANDING_PAGE
```

Comment:
@timabbott I am not sure whether this check is necessary. We can add no index to all doc pages as don't want these pages to be indexed by Google anyway(even in the case of root domain). But since the /integration pages are populated by content of these we don't want the integration pages to inherit no index from doc pages in root domainl. That is why I have included the check.

But it seems like none of the integration pages are indexed by Google (Probably due to the dynamic generation of these pages). So this check doesn't matter either anyway as these are already in no-index state due to the dynamic generation.

Reply:
I think they're not indexed by Google just because they aren't linked to.

I agree that for the /integrations/doc, we probably don't need this conditional; we don't want folks landing on these pages, since by design, they are HTML fragments.

I'm not sure I like the model of adding the `<meta>` tag, though; those belong in the `<head>` section of an HTML document, etc.

I think what we want to do instead is just have this endpoint be more explicitly configured to only return anything at all if it's an AJAX request, and e.g. just 404 for Google (etc)?